torch/csrc/jit/codegen/cuda/tensor_view.cpp

```diff
@@ -1310,27 +1313,53 @@ TensorViewBuilder& TensorViewBuilder::shape(std::vector<Val*> shape) {
   return *this;
 }
 
+TensorViewBuilder& TensorViewBuilder::expanded(std::vector<bool> expanded) {
+  TORCH_CHECK(expanded_.empty(), "Attempting to reset expanded shape");
+  if (!expanded.empty()) {
+    TORCH_CHECK(ndims_ == 0 || ndims_ == expanded.size());
+    ndims_ = expanded.size();
+  }
+  expanded_ = std::move(expanded);
+  return *this;
+}
+
 TensorView* TensorViewBuilder::build() const {
   // Build the domain
   std::vector<IterDomain*> domain(ndims_, nullptr);
   for (const auto i : c10::irange(ndims_)) {
+    bool is_expanded = false;
+    Val* extent = nullptr;
+    Val* expanded_extent = nullptr;
+    Val** shape_extent = &extent;
+    if (!expanded_.empty()) {
+      is_expanded = expanded_[i];
+    }
+    if (is_expanded) {
+      extent = FusionGuard::getCurFusion()->oneVal();
+      shape_extent = &expanded_extent;
+    }
     if (shape_.empty()) {
-      domain[i] =
-          IterDomainBuilder(
-              FusionGuard::getCurFusion()->zeroVal(), IrBuilder::create<Int>())
-              .build();
+      *shape_extent = IrBuilder::create<Int>();
     } else {
-      if (shape_[i]->isOneInt()) {
-        // If size is known to be 1, assume it needs to be broadcasted.
-        domain[i] = IterDomainBuilder(
-                        FusionGuard::getCurFusion()->zeroVal(),
-                        FusionGuard::getCurFusion()->oneVal())
-                        .iter_type(IterType::Broadcast)
-                        .build();
-      } else {
-        domain[i] =
-            IterDomainBuilder(FusionGuard::getCurFusion()->zeroVal(), shape_[i])
-                .build();
+      *shape_extent = shape_[i];
+    }
+    IterDomainBuilder builder(FusionGuard::getCurFusion()->zeroVal(), extent);
```

Comment:
Should we do some sanity checks? If expanded is true, the extent should be 1, right?

Reply:
Isn't this redundant with the above code?
```C++
    if (is_expanded) {
      extent = FusionGuard::getCurFusion()->oneVal();
      shape_extent = &expanded_extent;
    }
```

Reply:
OK, then that seems fine. How about an expanded domain with no shape setting? We don't allow symbolic expanded extents, do we?

Reply:
We do support symbolic expanded extents. I added a commit to this PR https://github.com/csarofeen/pytorch/pull/2049/commits/3bacbac57ef610efee60ab741a5f940fe16b625a to switch my test to symbolic expanded extents. However, the shape inference in the executor seems to not support input tensors with expanded extents well, so I have to manually make the extent an input of the fusion to make it work. We can fix this issue in a followup PR.

Reply:
OK, didn't know that. Approved the PR.